### PR TITLE
Azure Attestation create, get and delete api

### PR DESCRIPTION
**Pull request by DevOnGekko**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process.

### PR information
- [ ] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [ ] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [ ] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [ ] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [ ] My spec meets the review criteria:
  - [ ] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [ ] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Comment by dsgouda** [comment]
@DevOnGekko 
please confirm that these APIs have been through ARM and API reviews

**Comment by dsgouda** [comment]
@DevOnGekko Also fix all the CI failures reported

**Review by dsgouda — CHANGES_REQUESTED** [review]
This is a partial review, will review the rest when semantic errors are fixed and ARM/API review is finished

File: specification/azureattestation/resource-manager/Microsoft.AzureAttestation/preview/2017-11-01/AzureAttestation.json [hunk 1]
```diff
@@ -0,0 +1,433 @@
...
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/AttestationSigningKeys"
+            }
+          },
+          "404": {
```
[the remark this answers is not in the document]
Comment (DevOnGekko) [reply specification/azureattestation/resource-manager/Microsoft.AzureAttestation/preview/2017-11-01/AzureAttestation.json]: Fixed

File: specification/azureattestation/resource-manager/Microsoft.AzureAttestation/preview/2017-11-01/AzureAttestation.json [hunk 2]
```diff
@@ -0,0 +1,433 @@
...
+        }
+      }
+    },
+    "SigningKey": {
+      "type": "object",
+      "description": "Attestation signing keys.",
+      "properties": {
+        "kid": {
+          "type": "string",
+          "description": "Attestation signing keys."
```
[the remark this answers is not in the document]
Comment (DevOnGekko) [reply specification/azureattestation/resource-manager/Microsoft.AzureAttestation/preview/2017-11-01/AzureAttestation.json]: fixed

File: specification/azureattestation/resource-manager/Microsoft.AzureAttestation/preview/2017-11-01/AzureAttestation.json [hunk 3]
```diff
@@ -0,0 +1,433 @@
...
+            "description": "Created or updated attestation service",
+            "schema": {
+              "$ref": "#/definitions/AttestationResponseMessage"
+            }
+          },
+          "default": {
+            "description": "Error result from Attestation service",
+            "schema": {
+              "$ref": "#/definitions/ErrorResult"
+            }
```
[the remark this answers is not in the document]
Comment (DevOnGekko) [reply specification/azureattestation/resource-manager/Microsoft.AzureAttestation/preview/2017-11-01/AzureAttestation.json]: Its not long running taken few ms

File: specification/azureattestation/resource-manager/Microsoft.AzureAttestation/preview/2017-11-01/AzureAttestation.json [hunk 4]
```diff
@@ -0,0 +1,433 @@
...
+        ],
+        "responses": {
+          "202": {
+            "description": "Request accepted for deletion of attestation service"
+          },
+          "default": {
+            "description": "Error result from Attestation service",
+            "schema": {
+              "$ref": "#/definitions/ErrorResult"
+            }
```
[the remark this answers is not in the document]
Comment (DevOnGekko) [reply specification/azureattestation/resource-manager/Microsoft.AzureAttestation/preview/2017-11-01/AzureAttestation.json]: Not long running task takes few ms

File: specification/azureattestation/resource-manager/Microsoft.AzureAttestation/preview/2017-11-01/AzureAttestation.json [hunk 5]
```diff
@@ -0,0 +1,433 @@
...
+		"properties":{
+		  "type": "object",
+          "$ref": "#/definitions/StatusResult",
+          "description": "Describes Attestation service status.",
+          "x-ms-client-flatten": true
+		}
+      }
+    },
+    "StatusResult": {
+      "type": "object",
```
[the remark this answers is not in the document]
Comment (DevOnGekko) [reply specification/azureattestation/resource-manager/Microsoft.AzureAttestation/preview/2017-11-01/AzureAttestation.json]: Thanks!, its a Get operation hence changed the signature

File: specification/azureattestation/resource-manager/Microsoft.AzureAttestation/preview/2017-11-01/AzureAttestation.json [hunk 6]
```diff
@@ -0,0 +1,444 @@
...
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/AttestationSigningKeys"
```
Comment (dsgouda) [inline specification/azureattestation/resource-manager/Microsoft.AzureAttestation/preview/2017-11-01/AzureAttestation.json]: Please add the `x-ms-pageable` extension here

File: specification/azureattestation/resource-manager/Microsoft.AzureAttestation/preview/2017-11-01/AzureAttestation.json [hunk 7]
```diff
@@ -0,0 +1,444 @@
...
+      }
+    }
+  },
+  "paths": {
+    "/providers/Microsoft.AzureAttestation/operations": {
+      "get": {
+        "tags": [
+          "Operations"
+        ],
+        "operationId": "Operations_List",
```
Comment (dsgouda) [inline specification/azureattestation/resource-manager/Microsoft.AzureAttestation/preview/2017-11-01/AzureAttestation.json]: Make this operation as pageable

**Review by dsgouda — CHANGES_REQUESTED** [review]
Left a few more comments

**Comment by dsgouda** [comment]
@azuresdkci Retest this please

File: specification/azureattestation/resource-manager/Microsoft.AzureAttestation/preview/2017-11-01/AzureAttestation.json [hunk 8]
```diff
@@ -0,0 +1,444 @@
...
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.AzureAttestation/service/{serviceName}":
+    {
+      "get": {
+        "tags": [
+          "AttestationService"
+        ],
+        "operationId": "Get_AttestationService",
```
Comment (dsgouda) [inline specification/azureattestation/resource-manager/Microsoft.AzureAttestation/preview/2017-11-01/AzureAttestation.json]: OperationIds must be of the form `Resource_OperationName`, please take a look at rules R1003-R1007 [here](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/openapi-authoring-automated-guidelines.md) and rename operation Ids accordingly

**Review by dsgouda — CHANGES_REQUESTED** [review]

**Comment by dsgouda** [comment]
Closing this in lieu of PR opened [here](https://github.com/Azure/azure-rest-api-specs-pr/pull/551)

**Pull request closed without merging**